packages/flutter_tools/lib/src/commands/precache.dart

```diff
@@ -61,6 +63,9 @@ class PrecacheCommand extends FlutterCommand {
       if (argResults['macos']) {
         requiredArtifacts.add(DevelopmentArtifact.macOS);
       }
+      if (argResults['fuchsia']) {
```

Comment:
How hard would it be to split off the web, linux, etc. artifacts into an UnstableDevelopmentArtifacts? Then you could do something like.
```
for (DevelopmentArtifact artifact in DevelopmentArtifacts.values) {
  if (argResults[artifact.toString()]) {
    requiredArtifacts.add(artifact);
  }
}
if (!FlutterVersion.instance.isStable) {
  for (UnstableDevelopmentArtifact artifact in UnstableDevelopmentArtifacts.values) {
    if (argResults[artifact.toString()]) {
      requiredArtifacts.add(artifact);
    }
  }
}
```
Or maybe put the unstable development artifacts beyond some dummy marker in DevelopmentArtifacts and then iterate over the indices of DevelopmentArtifacts.values before/after that marker.

Reply:
Updated DevelopmentArtifacts to be a class with a name/unstable marker, ptal?